src/MicrosoftTeams.ts

```diff
@@ -795,6 +836,33 @@ namespace microsoftTeams {
             }
         }
 
+        /*
+        * Validates that the state param is a valid url with host as outlook.office.com and client_type correctly set to Outlook Desktop
+        * @param uri - the url to validate
+        */
+        function isValidState(state: string): boolean {
+            let link = document.createElement("a");
+            link.href = state;
+            if (link.host && link.host !== window.location.host && link.host === "outlook.office.com" && link.search.indexOf("client_type=Win32_Outlook") > -1) {
+                return true;
+            }
+            return false;
+        }
+
+        /**
+         * Appends either result or reason as a hash param to the 'state' url 
+         * @param uri - the url to modify
+         * @param key - the QSP key
+         * @param value - the QSP value
+         */
+        function updateUrlParameter(uri: string, key: string, value: string): string {
+            let i = uri.indexOf("#");
+            let hash = i === -1 ? "" : uri.substr(i);
+            hash = hash + "&" + key + "=" + value;
```

Comment:
If there is no # in the uri this code will just append a new QSP - &<key>=<value> - whereas if there is a # in the uri the key-value pair will be added after it which has completely different behavior. Is this intentional? Maybe the "" should actually be "#"?

Reply:
You're right. It should be "#". Will update in the next iteration.